src/diem/identifier/__init__.py

```diff
@@ -66,13 +66,17 @@ def account_id(self) -> str:
         return encode_account(self.account_address, self.sub_address, self.hrp)
 
 
-def encode_intent(encoded_account_identifier: str, currency_code: str, amount: int) -> str:
+def encode_intent(
+    encoded_account_identifier: str, currency_code: typing.Optional[str] = None, amount: typing.Optional[int] = None
+) -> str:
     """
     Encode account identifier string(encoded), currency code and amount into
     Diem intent identifier (https://dip.diem.com/dip-5/)
     """
 
-    return "diem://%s?c=%s&am=%d" % (encoded_account_identifier, currency_code, amount)
+    if currency_code and amount:
```

Comment:
shouldn't it also handle cases where only one of currency code or amount is not none?